backend/app/pyproject.toml

```diff
@@ -0,0 +1,85 @@
+[tool.poetry]
+name = "backend"
+version = "0.0.0"
+description = ""
+authors = []
+license = "AGPL-3.0"
+
+[tool.poetry.dependencies]
+python = "^3.8"
+alembic = "^1.4.2"
+Click = "^7.1.2"
+fastapi = "^0.56.0"
+gunicorn = "^20.0.4"
+h11 = "^0.9.0"
+httptools = "^0.1.1"
+jinja2 = "^2.11.2"
+mako = "^1.1.3"
+markupsafe = "^1.1.1"
+psycopg2-binary = "^2.8.5"
+pydantic = "^1.5.1"
+starlette = "0.13.2"
+tenacity = "^6.2.0"
+uvicorn = "^0.11.5"
+
+[tool.poetry.dev-dependencies]
```

Comment:
Do these dependency lists include things that are not top-level dependencies, and do we want to keep it that way? I feel like we really only get the full benefit of poetry if we minimize the practice of declaring the versions for dependencies of dependencies.

Reply:
That's a good point. Let me do some more testing and see if I can get it to only install top level dependencies. I had only fastapi and starlette above and i was getting errors that it didn't have uvicorn. This gave me the impression that poetry installs ONLY what you tell it.

Reply:
```
$ poetry show -t --no-dev
alembic 1.4.2 A database migration tool for SQLAlchemy.
├── mako *
│   └── markupsafe >=0.9.2 
├── python-dateutil *
│   └── six >=1.5 
├── python-editor >=0.3
└── sqlalchemy >=1.1.0
click 7.1.2 Composable command line interface toolkit
fastapi 0.56.0 FastAPI framework, high performance, easy to learn, fast to code, ready for production
├── pydantic >=0.32.2,<2.0.0
└── starlette 0.13.2
gunicorn 20.0.4 WSGI HTTP Server for UNIX
└── setuptools >=3.0
h11 0.9.0 A pure-Python, bring-your-own-I/O implementation of HTTP/1.1
httptools 0.1.1 A collection of framework independent HTTP protocol utils.
jinja2 2.11.2 A very fast and expressive template engine.
└── markupsafe >=0.23
mako 1.1.3 A super-fast templating language that borrows the  best ideas from the existing templating languages.
└── markupsafe >=0.9.2
markupsafe 1.1.1 Safely add untrusted strings to HTML/XML markup.
psycopg2-binary 2.8.5 psycopg2 - Python-PostgreSQL Database Adapter
pydantic 1.5.1 Data validation and settings management using python 3.6 type hinting
python-dateutil 2.8.1 Extensions to the standard Python datetime module
└── six >=1.5
six 1.15.0 Python 2 and 3 compatibility utilities
starlette 0.13.2 The little ASGI library that shines.
tenacity 6.2.0 Retry code until it succeeds
└── six >=1.9.0
uvicorn 0.11.5 The lightning-fast ASGI server.
├── click >=7.0.0,<8.0.0
├── h11 >=0.8,<0.10
├── httptools >=0.1.0,<0.2.0
├── uvloop >=0.14.0
└── websockets >=8.0.0,<9.0.0
```
poetry does not seem to think uvicorn is a dependency of fastapi or starlette

Reply:
It might be a peer dependency rather than a child?

Reply:
ah that makes sense. i don't know why i thought it was a part those. as i rack my brain i remember it's added to the project b/c uvloop is a much faster event loop implementation for asyncio.